pallets/interest-accrual/src/lib.rs

```diff
@@ -335,30 +335,35 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub fn get_current_debt(
-			interest_rate_per_year: T::InterestRate,
-			normalized_debt: T::Balance,
-		) -> Result<T::Balance, DispatchError> {
-			let rate = Self::get_rate(interest_rate_per_year)?;
-			Self::calculate_debt(normalized_debt, rate.accumulated_rate)
-				.ok_or_else(|| Error::<T>::DebtCalculationFailed.into())
-		}
-
-		pub fn get_previous_debt(
+		/// Calculate fastly the current debt using normalized debt * cumulative rate if
+		/// `when` is exactly `now` (same block). If when is in the past it recomputes
+		/// the previous cumulative rate.
+		///
+		/// If `when` is further in the past than the last time the
+		/// normalized debt was adjusted, this will return nonsense
+		/// (effectively "rewinding the clock" to before the value was
+		/// valid)
+		pub fn get_debt(
 			interest_rate_per_year: T::InterestRate,
 			normalized_debt: T::Balance,
 			when: Moment,
 		) -> Result<T::Balance, DispatchError> {
 			let rate = Self::get_rate(interest_rate_per_year)?;
 			let now = LastUpdated::<T>::get();
-			if when > now {
-				return Err(Error::<T>::NotInPast.into());
-			}
-			let delta = now - when;
-			let rate_adjustment = checked_pow(rate.interest_rate_per_sec, delta.ensure_into()?)
-				.ok_or(ArithmeticError::Overflow)?;
-			let past_rate = rate.accumulated_rate.ensure_div(rate_adjustment)?;
-			Self::calculate_debt(normalized_debt, past_rate)
+
+			let acc_rate = match when.cmp(&now) {
+				Ordering::Equal => rate.accumulated_rate,
+				Ordering::Less => {
+					let delta = now.ensure_sub(when)?;
+					let rate_adjustment =
+						checked_pow(rate.interest_rate_per_sec, delta.ensure_into()?)
+							.ok_or(ArithmeticError::Overflow)?;
+					rate.accumulated_rate.ensure_div(rate_adjustment)?
```

Comment:
Minor: but I like 

`Match when.cmp(now)`

Makes reading easier IMO and control flow safer.

Reply:
Yeah, I agree 🤣